pytorch_lightning/core/optimizer.py

```diff
@@ -337,15 +337,26 @@ def _get_default_scheduler_config() -> Dict[str, Any]:
         "reduce_on_plateau": False,  # most often not ReduceLROnPlateau scheduler
         "monitor": None,  # value to monitor for ReduceLROnPlateau
         "strict": True,  # enforce that the monitor exists for ReduceLROnPlateau
-        "opt_idx": None,  # necessary to store opt_idx when optimizer frequencies are specified
+        "opt_idx": None,  # opt_idx assigned internally if not assigned by user
```

Comment:
Should this default to 0 since you removed `Optional` from the annotation in `types.py`?

```suggestion
        "opt_idx": 0,  # opt_idx assigned internally if not assigned by user
```

Reply:
it's required to differentiate from the opt_idx provided by the user in scheduler config and based on your below comment it should stay None

Reply:
I agree. Then you should revert the change to `types.py`

Reply:
just clarifying.. if we update it eventually to an int for all schedulers, won't it be int in `types.py`?? cause we use this config just as an initial entry point.

Reply:
If the config ever allows None, it should be `Optional[int]`.

There are complex tricks to tell mypy that the type has been narrowed down later (not saying you should do this here): https://www.python.org/dev/peps/pep-0647/

Reply:
thanks :)